include/seqan3/alignment/matrix/alignment_trace_matrix.hpp

```diff
@@ -98,29 +99,108 @@ struct alignment_coordinate
     size_t seq2_pos;
 };
 
-/*!\brief Compute the trace from a trace matrix
+/*!\brief Compute the number of insertions, deletions and
+ * substitutions between begin coordinate and end coordinate.
  * \ingroup alignment_matrix
- * \tparam    database_t     The type of the database sequence.
- * \tparam    query_t        The type of the query sequence.
  * \tparam    trace_matrix_t The type of the trace matrix.
- * \param[in] database       The database sequence.
- * \param[in] query          The query sequence.
  * \param[in] matrix         The trace matrix.
  * \param[in] end_coordinate Where the trace in the matrix ends.
  */
+template <typename trace_matrix_t>
+//!\cond
+    requires matrix_concept<remove_cvref_t<trace_matrix_t>> &&
+             std::Same<typename remove_cvref_t<trace_matrix_t>::entry_type, trace_directions>
+//!\endcond
+inline std::tuple<size_t, size_t, size_t>
+alignment_trace_statistics(trace_matrix_t && matrix, alignment_coordinate const end_coordinate)
+{
+    using signed_size_t = std::make_signed_t<size_t>;
+    constexpr auto N = trace_directions::none;
+    constexpr auto D = trace_directions::diagonal;
+    constexpr auto L = trace_directions::left;
+    constexpr auto U = trace_directions::up;
+    size_t col = end_coordinate.seq1_pos + 1;
+    size_t row = end_coordinate.seq2_pos + 1;
+    size_t inserts = 0;
+    size_t deletes = 0;
+    size_t substitutes = 0;
+
+    while(row >= 0 && col >= 0)
+    {
+        trace_directions dir = matrix.at(row, col);
+        if ((dir & L) == L)
+        {
+            col = std::max<signed_size_t>(col - 1, 0);
+            ++inserts;
+        }
+        else if ((dir & U) == U)
+        {
+            row = std::max<signed_size_t>(row - 1, 0);
+            ++deletes;
+        }
+        else if ((dir & D) == D)
+        {
+            row = std::max<signed_size_t>(row - 1, 0);
+            col = std::max<signed_size_t>(col - 1, 0);
+            ++substitutes;
+        }
+        else
+        {
+            if (row == 0 || col == 0)
+                break;
+            throw std::logic_error{"Trace not found"};
+        }
+    }
+    return {inserts, deletes, substitutes};
+}
+
+/*!\brief Compute the begin coordinate.
+ * \ingroup alignment_matrix
+ * \tparam    trace_matrix_t The type of the trace matrix.
+ * \param[in] end_coordinate Where the trace in the matrix ends.
+ * \param[in] statistics     Statistics returned by alignment_trace_statistics.
+ */
+inline alignment_coordinate
+alignment_begin_coordinate(alignment_coordinate const end_coordinate, std::tuple<size_t, size_t, size_t> const & statistics)
```

Comment:
Is this a function only needed in combination with statistics? otherwise, you might want to add a function that takes the end coordinate and the matrix and computes the begin position

Reply:
changed this :)